patches/minecraft/net/minecraft/world/entity/npc/VillagerTrades.java.patch

```diff
@@ -0,0 +1,21 @@
+--- a/net/minecraft/world/entity/npc/VillagerTrades.java
++++ b/net/minecraft/world/entity/npc/VillagerTrades.java
+@@ -1367,7 +_,7 @@
+         private final int villagerXp;
+ 
+         public EmeraldsForVillagerTypeItem(int p_35669_, int p_35670_, int p_35671_, Map<VillagerType, Item> p_35672_) {
+-            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)).findAny().ifPresent(p_327046_ -> {
++            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_) && BuiltInRegistries.VILLAGER_TYPE.getKey(p_35680_).getNamespace().equals("minecraft")).findAny().ifPresent(p_327046_ -> {
```

Comment:
This is more of a nitpick than anything else, but moving the namespace check to its own line in the patch makes it easier to distinguish what's actually been changed. Also, add a forge comment.

```suggestion
-            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)).findAny().ifPresent(p_327046_ -> {
+            // FORGE: Only check vanilla villager types so registering modded types doesn't crash
+            BuiltInRegistries.VILLAGER_TYPE.stream().filter(p_35680_ -> !p_35672_.containsKey(p_35680_)
+            && BuiltInRegistries.VILLAGER_TYPE.getKey(p_35680_).getNamespace().equals("minecraft")).findAny().ifPresent(p_327046_ -> {
```